swebench_docker/run_docker.py

```diff
@@ -30,45 +28,64 @@ async def run_docker_evaluation(task_instance: dict, namespace: str, log_dir: st
     else:
         docker_image = f"{namespace}/{image_prefix}-{repo_name}-testbed:{task_instance['version']}"
 
-    docker_command = [
-        'docker', 'run',
-        '--rm',
-        '-v', f"{log_dir}:{container_log_dir}",
-        '-e', f"INSTANCE={instance_b64}",
-        '-e', f"LOG_DIR={container_log_dir}",
-        '-e', f"TIMEOUT={timeout}",
-        '-e', f"LOG_SUFFIX={log_suffix}",
-        docker_image
-    ]
-
-    cmd_string = ' '.join(docker_command)
-
-    if verbose:
-        logger.info(cmd_string)
-
-    start_time = time.time()
-
-    try:
-        process = await asyncio.create_subprocess_shell(cmd_string, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
-        stdout, stderr = await process.communicate()
-        stdout = stdout.decode()
-        if stderr:
-            stderr = stderr.decode()
-
-        elapsed_time = time.time() - start_time
-
-        if process.returncode != 0:
-            logger.warning(
-                f"[{task_instance['instance_id']}][{docker_image}]  Error running container:")
-            logger.warning(f"Command: {cmd_string}")
-            logger.warning(f"Stdout - {stdout}")
-            logger.warning(f"Stderr - {stderr}")
-
-        elif "Evaluation succeeded" not in stdout:
-            logger.warning(f"[{task_instance['instance_id']}][{docker_image}]  Container ran successfully in {elapsed_time} seconds, but evaluation failed.")
-            logger.warning(f"Command: {cmd_string}")
-            logger.warning(f"stdout - {stdout}")
-        else:
-            logger.info(f"[{task_instance['instance_id']}][{docker_image}] Container ran successfully in {elapsed_time} seconds.")
-    except Exception as e:
-        logger.warning(f"[{task_instance['instance_id']}][{docker_image}]  Error running container: {e}")
+    # Create a temporary file to store the task_instance JSON
+    with tempfile.NamedTemporaryFile(delete=False, mode='w+', suffix='.json') as tmpfile:
+        json.dump(task_instance, tmpfile)
+        tmpfile.close()
+        tmpfile_path = tmpfile.name
+
+        swebench_docker_fork_dir = os.environ.get("SWEBENCH_DOCKER_FORK_DIR")
+        docker_command = [
+            'docker', 'run',
+            '--rm',
+            '-v', f"{log_dir}:{container_log_dir}",
+            # Map the swebench_docker fork dir to the container
+            # for some reason, swebench_docker has different locations for the different containers :(
```

Comment:
Maybe we should just change the docker generation scripts on the pyenv docker images to also use `/home/swe-bench` instead of `/opt/`?

Reply:
Oh yes - That would be ideal! But I'm hoping we can do that in a separate PR since we will need to re-build all the containers and make sure they works -- I'll probably try to tackle this in the next week or so!